domain/cloud/state/state.go

```diff
@@ -73,6 +75,234 @@ func (st *State) ListClouds(ctx context.Context, filter *Filter) ([]cloud.Cloud,
 	return result, errors.Trace(err)
 }
 
+// CloudDefaults provides the currently set cloud defaults for a cloud. If the
+// cloud has no defaults or the cloud does not exist a nil error is returned
+// with an empty defaults map.
+func (st *State) CloudDefaults(ctx context.Context, cloudName string) (map[string]string, error) {
+	defaults := map[string]string{}
```

Comment:
I wonder if it's better to return a nil map instead of an empty one if the cloud doesn't exist? (we return nil in various places elsewhere for this type of case). I guess empty map avoids potential nil pointer errors, and this is new so maybe consistency isn't an issue (yet).

Reply:
I am happy to go with the flow. I think from our discussions we will leave it as an empty map for now as it adds some defensive programming. We can always rejig later if we decide.